packages/@aws-cdk/aws-iotevents/lib/DetectorModel.ts

```diff
@@ -0,0 +1,310 @@
+  /**
+   * Name for the {@link DetectorModel}
+   *
+   * @type {string}
+   * @memberof IDetectorModel
+   * @attribute
+   */
+  readonly detectorModelName?: string;
+
+  /**
+   * The IAM service Role associated with this detector model.
+   * This role is used as argument for {@link CfnDetectorModel} to
+   * "grant permission to AWS IoT Events to perform its operations".
+   *
+   * @type {IRole}
+   * @memberof IDetectorModel
+   */
+  readonly role?: IRole;
+  /**
+   * The grant principal for this Detector Model
+   *
+   * @type {IPrincipal}
+   * @memberof IDetectorModel
+   */
+  readonly grantPrincipal?: IPrincipal;
+  /**
+   * Adds a statement to the IAM role assumed by the detector model
+   *
+   * @param {PolicyStatement} policyStatement
+   * @memberof IDetectorModel
+   */
+  addToRolePolicy(policyStatement: PolicyStatement): void;
+}
+
+/**
+ * Detector model attributes for import
+ *
+ * @export
+ * @interface DetectorModelAttributes
+ */
+export interface DetectorModelAttributes {
+  /**
+   * Arn for the model
+   *
+   * @type {string}
+   * @memberof DetectorModelAttributes
+   */
+  readonly detectorModelArn: string;
+
+  /**
+   * Name of the model
+   *
+   * @type {string}
+   * @memberof DetectorModelAttributes
+   * @default - generated from Arn
+   */
+  readonly detectorModelName?: string;
+
+  /**
+   * Arn for the role of the model
+   *
+   * @type {string}
+   * @memberof DetectorModelAttributes
+   * @default - no role
+   */
+  readonly roleArn?: string;
+}
+/**
+ * Base class for an {@link DetectorModel}
+ *
+ * @export
+ * @abstract
+ * @class DetectorModelBase
+ * @extends {Resource}
+ * @implements {IDetectorModel}
+ */
+abstract class DetectorModelBase extends Resource implements IDetectorModel {
+  public abstract readonly detectorModelArn: string;
+
+  public abstract readonly detectorModelName?: string;
+  public abstract readonly grantPrincipal?: IPrincipal;
+  public abstract readonly role?: IRole;
+
+  public addToRolePolicy(policyStatement: PolicyStatement): void {
+    if (this.role) {
+      this.role.addToPolicy(policyStatement);
+    }
+  }
+}
+/**
+ * Props for Detector Model
+ *
+ * @export
+ * @interface DetectorModelProps
+ */
+export interface DetectorModelProps {
+  /**
+   * Human readable description for the model
+   *
+   * @type {string}
+   * @memberof DetectorModelProps
+   * @default - no description
+   */
+  readonly description?: string;
+  /**
+   * Optional name for the model
+   *
+   * @type {string}
+   * @memberof DetectorModelProps
+   * @default - generated name
+   */
+  readonly detectorModelName?: string;
+  /**
+   * Service Role to assume while running the build.
+   *
+   * @default - A role will be created.
+   */
+  readonly role?: IRole;
+  /**
+   * The initial state
+   *
+   * @type {IState}
+   * @memberof DetectorModelProps
+   * @default - empty model
+   */
+  readonly entryPoint?: IState;
+
+  /**
+   * The key separating detectors
```

Comment:
I'm assuming this key has nothing to do with KMS, but it'd be good to add a bit more detail to further disambiguate the context in which `key` will be used